lib/fdiff/__main__.py

```diff
@@ -63,40 +64,42 @@ def run(argv):
         "--nomp", action="store_true", help="Do not use multi process optimizations"
     )
     parser.add_argument("--external", type=str, help="Run external diff tool command")
-    parser.add_argument("PREFILE", help="Font file path/URL 1")
-    parser.add_argument("POSTFILE", help="Font file path/URL 2")
 
-    args = parser.parse_args(argv)
+    parser.add_argument("--git", type=str, nargs=7, help="Act as a diff driver for git (takes 7 parameters)")
+    #  parser.add_argument("PREFILE", help="Font file path/URL 1")
+    #  parser.add_argument("POSTFILE", help="Font file path/URL 2")
+
+    args, positionals = parser.parse_known_args(argv)
+
+    inputs = argparse.Namespace()
+    if args.git:
+        inputs.PREFILE = args.git[1]
+        inputs.POSTFILE = args.git[4]
+    else:
+        inputparser = argparse.ArgumentParser()
+        inputparser.add_argument("PREFILE", help="Font file path/URL 1")
+        inputparser.add_argument("POSTFILE", help="Font file path/URL 2")
+        inputparser.parse_args(positionals, namespace=inputs)
+
 
     # /////////////////////////////////////////////////////////
     #
     #  Validations
     #
     # /////////////////////////////////////////////////////////
 
-    # ----------------------------------
-    #  Incompatible argument validations
-    # ----------------------------------
-    #   --include and --exclude are mutually exclusive options
-    if args.include and args.exclude:
```

Comment:
Do you get a reasonably informative message for the exclusive options when you switch to the argparse automation?

Reply:
Yes. The wording is a little different that your custom one but it does explain the situation clearly if you pass the wrong ags.

Reply:
> fdiff: error: argument --exclude: not allowed with argument --include